src/openstudio_lib/MainWindow.cpp

```diff
@@ -285,12 +294,17 @@ void MainWindow::writeSettings() {
   settings.setValue("displayIP", m_displayIP);
   settings.setValue("verboseOutput", m_verboseOutput);
   settings.setValue("language", m_currLang);
+  settings.setValue("analyticsId", m_analyticsId);
 }
 
 QString MainWindow::lastPath() const {
   return QDir().exists(m_lastPath) ? m_lastPath : QDir::homePath();
 }
 
+bool MainWindow::allowAnalytics() const {
+  return !m_analyticsId.isEmpty() && m_analyticsId != "DISABLED";
+}
```

Comment:
Ok so that's why you're not using a bool, because you would actually need a tristate boolean or an `optional<bool>` to check whether initialized or not.